src/go/rdctl/cmd/root.go

```diff
@@ -107,9 +126,32 @@ func getRequestObject(method string, command string) (*http.Request, error) {
 	return req, nil
 }
 
-func doRestOfRequest(req *http.Request) ([]byte, error) {
+func doRestOfRequest(req *http.Request) (*http.Response, error) {
 	client := http.Client{}
-	response, err := client.Do(req)
+	return client.Do(req)
+}
+
+func processRequestForAPI(response *http.Response, err error) ([]byte, *APIError, error) {
+	if err != nil {
+		return nil, nil, err
+	}
+	errorPacket := APIError{}
+	pErrorPacket := &errorPacket
+	if response.StatusCode < 200 || response.StatusCode >= 300 {
+		errorPacket.Message = &response.Status
+	} else {
+		pErrorPacket = nil
+	}
+	defer response.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, pErrorPacket, nil
```

Comment:
So if we fail to read response body (e.g. HTTP 200, but the connection was dropped halfway through), we'd end up returning `nil, nil, nil`? That doesn't seem right…

Reply:
Good catch. It's a bit messy, having two formatters, one for the `api` command and the other for the other "utility" commands, but I overlooked that with the api formatter.